packages/core/src/hydration/event_replay.ts

```diff
@@ -185,23 +194,60 @@ export function setJSActionAttribute(
   tNode: TNode,
   rNode: RNode,
   nativeElementToEvents: Map<Element, string[]>,
+  parentDeferBlockId: string | null = null,
 ) {
   if (tNode.type & TNodeType.Element) {
     const nativeElement = unwrapRNode(rNode) as Element;
     const events = nativeElementToEvents.get(nativeElement) ?? [];
-    const parts = events.map((event) => `${event}:`);
+    const namespace = parentDeferBlockId ?? '';
+    const parts = events.map((event) => `${event}:${namespace}`);
```

Comment:
WDYM? What would be repeated?

Reply:
I was thikning instead of

jsaction="click:d1;focus:d1"

it'd be

jsaction-block="d1" jsaction="click;"

then we wouldn't have to regex

Reply:
It would be more bites over the wire to do that with the additional long attribute. Is there a desire to not use the action portion of the jsaction string?

Reply:
It's more that we do this work here as well (https://github.com/angular/angular/blob/01172b84d9e8ab37ea989783a480add80c442421/packages/core/primitives/event-dispatch/src/action_resolver.ts#L247) and it'd be nice to share that work (especially since it's cached here).